libcudacxx/include/cuda/__mdspan/host_device_accessor.h

```diff
@@ -202,25 +193,27 @@ class __host_accessor : public _Accessor
       : _Accessor{__acc}
   {}
 
-  _CCCL_API constexpr reference access(data_handle_type __p, size_t __i) const noexcept(__is_access_noexcept)
+  _CCCL_API constexpr reference access(data_handle_type __p, ::cuda::std::size_t __i) const
+    noexcept(__is_access_noexcept)
   {
-    NV_IF_ELSE_TARGET(
-      NV_IS_DEVICE,
-      (_CCCL_VERIFY(false, "cuda::__host_accessor cannot be used in DEVICE code");),
-      (_CCCL_ASSERT(__is_host_accessible_pointer(__p), "cuda::__host_accessor data handle is not a HOST pointer");))
+    NV_IF_TARGET(NV_IS_DEVICE, (_CCCL_VERIFY(false, "cuda::__host_accessor cannot be used in DEVICE code");))
     return _Accessor::access(__p, __i);
```

Comment:
Why are we dropping the check here? that seems like a regression

Reply:
Suggestion:  there are different ways to store that information during construction. We could either make it a boolean that is set during construction or turn the pointer into a tagged pointer which would be much more onvolved though